pyqtorch/modules/hamevo.py

```diff
@@ -166,46 +242,101 @@ def apply(self, state: torch.Tensor) -> torch.Tensor:
         return _apply_batch_gate(state, evol_operator, self.qubits, self.n_qubits, batch_size_h)
 
 
+class HamEvoType(Enum):
+    """
+    An Enumeration to represent types of Hamiltonian Evolution
+
+    RK4: Hamiltonian evolution performed using the 4th order Runge-Kutta method.
+    EIG: Hamiltonian evolution performed using Eigenvalue Decomposition.
+    EXP: Hamiltonian evolution performed using the Exponential of the Hamiltonian.
+    """
+
+    RK4 = HamEvo
+    EIG = HamEvoEig
+    EXP = HamEvoExp
+
+
 class HamiltonianEvolution(Module):
-    def __init__(self, qubits: Any, n_qubits: int, n_steps: int = 100):
+    """
+    A module to encapsulate Hamiltonian Evolution operations.
+
+    Performs Hamiltonian Evolution using different strategies
+    such as, RK4, Eigenvalue Decomposition, and Exponential,
+    Based on the 'hamevo_type' parameter.
+
+    Attributes:
+        qubits: A list of qubits to be used in the operation
+        n_qubits (int): Total number of qubits
+        n_steps  (int): The number of steps to be performed in RK4-based evolution. Defaults to 100
+        hamevo_type (Enum or str): The type of Hamiltonian evolution to be performed.
+                     Must be a member of the `HamEvoType` enum or equivalent string.
+
+    Examples:
+    (1)
+    # Instantiate HamiltonianEvolution with RK4 string input
+        >>> hamiltonian_evolution = HamiltonianEvolution(qubits, n_qubits, 100, "RK4")
+    # Use the HamiltonianEvolution instance to evolve the state
+        >>> output_state = hamiltonian_evolution(H, t, state)
+
+    (2)
+    # Instantiate HamiltonianEvolution with HamEvoType. input
+        >>>H_evol = HamiltonianEvolution(qubits, n_qubits, 100, HamEvoType.Eig)
+    # Use the HamiltonianEvolution instance to evolve the state
+        >>>output = H_evol(H, t, state) # Use the HamiltonianEvolution instance to evolve the state
+
+    """
+
+    def __init__(
+        self,
+        qubits: Any,
+        n_qubits: int,
+        n_steps: int = 100,
+        hamevo_type: Union[HamEvoType, str] = HamEvoType.RK4,
```

Comment:
@zain2864 Last minor comment: please set the default to `HamEvoType.HamEvoExp` since it is the most reliable method.